MIGRATING.md

```diff
@@ -0,0 +1,242 @@
++++ b/contracts/cw1-subkeys/Cargo.toml
+-cw0 = { path = "../../packages/cw0", version = "0.10.3" }
++utils = { path = "../../packages/utils", version = "0.10.3" }
+```
+
+```diff
+diff --git a/contracts/cw1-subkeys/src/contract.rs b/contracts/cw1-subkeys/src/contract.rs
+index b4852225..f20a65ec 100644
+--- a/contracts/cw1-subkeys/src/contract.rs
++++ b/contracts/cw1-subkeys/src/contract.rs
+-use cw0::Expiration;
++use utils::Expiration;
+```
+
+- Deprecate `range` to `range_raw` [\#460](https://github.com/CosmWasm/cw-plus/issues/460) /
+`range` to `range raw` [\#576](https://github.com/CosmWasm/cw-plus/pull/576).
+
+`range` was renamed to `range_raw` (no key deserialization), and `range_de` to `range`. This means you can now count
+on the key to be deserialized for you, which results in cleaner / simpler code.
+
+There are some examples in the contracts code base, by example:
+```diff
+diff --git a/contracts/cw3-fixed-multisig/src/contract.rs b/contracts/cw3-fixed-multisig/src/contract.rs
+index 48a60083..2f2ef70d 100644
+--- a/contracts/cw3-fixed-multisig/src/contract.rs
++++ b/contracts/cw3-fixed-multisig/src/contract.rs
+@@ -385,21 +384,20 @@ fn list_votes(
+     let limit = limit.unwrap_or(DEFAULT_LIMIT).min(MAX_LIMIT) as usize;
+     let start = start_after.map(Bound::exclusive);
+
+-    let votes: StdResult<Vec<_>> = BALLOTS
++    let votes = BALLOTS
+         .prefix(proposal_id)
+-        .range_raw(deps.storage, start, None, Order::Ascending)
++        .range(deps.storage, start, None, Order::Ascending)
+         .take(limit)
+         .map(|item| {
+-            let (key, ballot) = item?;
+-            Ok(VoteInfo {
+-                voter: String::from_utf8(key)?,
++            item.map(|(addr, ballot)| VoteInfo {
++                voter: addr.into(),
+                 vote: ballot.vote,
+                 weight: ballot.weight,
+             })
+         })
+-        .collect();
++        .collect::<StdResult<_>>()?;
+
+-    Ok(VoteListResponse { votes: votes? })
++    Ok(VoteListResponse { votes })
+ }
+
+ fn query_voter(deps: Deps, voter: String) -> StdResult<VoterResponse> {
+```
+
+If / when you don't need key deserialization, just can just rename `range` to `range_raw` and you are good to go.
+
+If you need / want key deserialization for **indexes**, you need to specify the primary key type as a last (optional)
+argument in the index key specification. If not specified, it defaults to `()`, which means that primary keys will
+not only not be deserialized, but also not provided. This is for backwards-compatibility with current indexes
+specifications, and may change in the future once these features are stabilized.
+See `packages/storage-plus/src/indexed_map.rs` tests for reference.
+
+Also, as part of this issue, `keys` was renamed to `keys_raw`, and `keys_de` to `keys`. `prefix_range` was also renamed,
+to `prefix_range_raw`, and its key deserialization counterpart (`prefix_range_de`) to `prefix_range` in turn.
```

Comment:
Hmm, I think that instead of fiddling with language to not sound repetetive maybe consider just list of renamed types?
```
Renamed types:
- keys -> keys_raw
- keys_de -> keys
- prefix_range -> prefix_range_raw
- prefix_range_de -> prefix_range
```